client/app/components/compose.coffee

```diff
@@ -162,24 +171,62 @@ module.exports = Compose = React.createClass
 
                 div className: 'clearfix', null
 
-    onCancel: (e) ->
-        e.preventDefault()
 
-        # Action after cancelation : call @props.onCancel
-        # or navigate to message list
-        if @props.onCancel?
-            @props.onCancel()
+    # Build message hash with the many ids required: account ID, mailbox ID,
```

Comment:
This whole block probably doesn't belong in a view component, the best way would be for the LayoutStore to generate and keep this "back url", and the panel component to pass it as a prop to the compose component.